amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFConfiguration.scala

```diff
@@ -27,42 +27,106 @@ import amf.core.internal.convert.PayloadValidationPluginConverter.PayloadValidat
 import amf.core.internal.remote.Spec
 import amf.shapes.client.platform.BaseShapesConfiguration
 
+/**
+  * The AMFConfiguration lets you customize all AMF-specific configurations.
+  * Its immutable and created through builders. An instance is needed to use AMF.
+  * @see [[AMFBaseUnitClient]]
+  */
 @JSExportAll
 class AMFConfiguration private[amf] (private[amf] override val _internal: scala.AMFConfiguration)
     extends BaseShapesConfiguration(_internal) {
 
+  /** Contains common AMF graph operations associated to documents */
   override def baseUnitClient(): AMFBaseUnitClient = new AMFBaseUnitClient(this)
-  override def elementClient(): AMFElementClient   = new AMFElementClient(this)
-  def configurationState(): AMFConfigurationState  = new AMFConfigurationState(this)
 
+  /** Contains functionality associated with specific elements of the AMF model */
+  override def elementClient(): AMFElementClient = new AMFElementClient(this)
+
+  def configurationState(): AMFConfigurationState = new AMFConfigurationState(this)
```

Comment:
In configurationState() you could add the same description that is in the class: /* Contains methods to get information about the current state of the configuration */